test/test_qml.cpp

```diff
@@ -72,23 +62,80 @@ class Setup : public QObject
                                                       << QStringLiteral( "tcp" )
                                                       << QStringLiteral( "--port" )
                                                       << QStringLiteral( "11111" )
-                                                      << QStringLiteral( "%1/happy.txt" ).arg( nmeaServer ) );
+                                                      << QStringLiteral( "%1/%2" ).arg( nmeaServer, mTcpTestFile ) );
         mTcpServerProcess.start();
 
         mUdpServerProcess.setProgram( QStringLiteral( "python3" ) );
         mUdpServerProcess.setArguments( QStringList() << QStringLiteral( "%1/nmeaserver.py" ).arg( nmeaServer )
                                                       << QStringLiteral( "--type" )
                                                       << QStringLiteral( "udp" )
-                                                      << QStringLiteral( "%1/TrimbleR1.txt" ).arg( nmeaServer ) );
+                                                      << QStringLiteral( "%1/%2" ).arg( nmeaServer, mUdpTestFile ) );
         mUdpServerProcess.start();
       }
     }
 
-    void cleanupTestCase()
+    void kill()
     {
       // kill the TCP and UDP server
       mTcpServerProcess.kill();
       mUdpServerProcess.kill();
+
+      if ( !mTcpServerProcess.waitForFinished() || !mUdpServerProcess.waitForFinished() )
+        qDebug() << "Vaiting for processes to terminate timed out";
+    }
+
+    void restart()
+    {
+      kill();
+      start();
+    }
+
+    QString dataDir() const { return mDataDir; }
+
+    void setTcpTestFile( const QString &fileName )
+    {
+      mTcpTestFile = fileName;
+    }
+
+    void setUdpTestFile( const QString &fileName )
```

Comment:
You'll need to make that Q_INVOKABLE if you want to call it from QML. A more QML-esque way would be to make that a Q_PROPERTY, but maybe it's an overkill for this specific need.

Reply:
Should not be enough that they are public slots ? Or because It has an argument? The other functions in each case gets called from qml

Reply:
What other functions are you referring to?

Reply:
The other that you commented, `restart`

Btw, is there an easy way to run the test locally?

Reply:
@domi4484 , why do you feel restart() works to begin with? :) 

Easiest way I found to run test locally is to build catch2 via a vcpkg setup, then manually set CATCH2_INCLUDE_DIRS etc. I've unfortunately reset my machine not too long ago and haven't re-enabled tests since then so my magic recipe is momentarily gone :/

Reply:
I am not sure anymore, too many tries at late night 🤔😁. But slot are practically the same things as Q_INVOKABLE. I have added some traces to be sure :)

Ok about running i'll continue let gthub doing for the moment.

Reply:
Seems the slot are called. But the server processes does't get stopped for some reason... A solution would probably be do fire up a third process on another udp port.
Or do you have an idea why they can't be stopped?

Reply:
Indeed, TIL. 

I can't test ATM, but yeah launching a third process would be an easy workaround in the meantime, until either you or I setup a proper testing environment on our machine.

Reply:
Ok running the tests is as simple as running `ctest` in the build folder 😀

- I have instaled catch2 from the distro repos
- Set ENABLE_TESTS in build config 
- Built QField locally whitout vcpkg
- Run ctest